File: 0204-count-primes-计数质数/0204.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n == 0) return 0;
        bool arr[n];
        for (int i = 0; i < n; ++i) arr[i] = true;
        int count = 0;
        for (int i = 2; i < n; i++) {
            if (arr[i] == true) {
                count += 1;
            }
            for (int j = 0; i * j < n; ++j) {
                arr[i * j] = false;
            }
        }
        return count;
    }
};

File: 0204-count-primes-计数质数/0204.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 2: return 0
        arr = [1 for _ in range(n+1)]
        count = 0
        for i in range(2, n):
            if arr[i] == 1:
                count += 1
            if i <= int(math.sqrt(n)):
                for j in range(i, n // i + 1):
                    arr[i*j] = 0
        
        return count


